pkg/disttask/framework/scheduler/scheduler.go

```diff
@@ -610,27 +615,29 @@ func (s *BaseScheduler) markSubTaskCanceledOrFailed(ctx context.Context, subtask
 		err := errors.Cause(err)
 		if ctx.Err() != nil && context.Cause(ctx) == ErrCancelSubtask {
 			logutil.Logger(s.logCtx).Warn("subtask canceled", zap.Error(err))
-			s.updateSubtaskStateAndError(subtask, proto.TaskStateCanceled, nil)
+			updateCtx := util.WithInternalSourceType(context.Background(), kv.InternalDistTask)
```

Comment:
tidb might not be able to shutdown gracefully if we block here
deadline might not work in all case, might we can use parent ctx, ok for now,

Reply:
fixed